protocol-designer/src/components/StepEditForm/types.ts

```diff
@@ -1,34 +1,26 @@
-// @flow
-import type { FormData, StepFieldName } from '../../form-types'
-
-export type FocusHandlers = {|
-  focusedField: StepFieldName | null,
-  dirtyFields: Array<StepFieldName>,
+import { FormData, StepFieldName } from '../../form-types'
+export interface FocusHandlers {
+  focusedField: StepFieldName | null
+  dirtyFields: StepFieldName[]
   // NOTE: focus & blur take a field name as an arg, unlike onFieldBlur/onFieldFocus in FieldProps
-  focus: StepFieldName => void,
-  blur: StepFieldName => void,
-|}
-
-export type FieldProps = {|
-  disabled: boolean,
-  errorToShow: ?string,
-  isIndeterminate?: boolean,
-  name: string,
-  onFieldBlur: () => mixed,
-  onFieldFocus: () => mixed,
-  tooltipContent?: ?string,
-  updateValue: mixed => void,
-  value: mixed,
-|}
-
-export type FieldPropsByName = {
-  [name: StepFieldName]: FieldProps,
-  ...
+  focus: (arg0: StepFieldName) => void
+  blur: (arg0: StepFieldName) => void
 }
-
+export interface FieldProps {
+  disabled: boolean
+  errorToShow?: string | null
+  isIndeterminate?: boolean
+  name: string
+  onFieldBlur: () => unknown
+  onFieldFocus: () => unknown
+  tooltipContent?: string | null | undefined
```

Comment:
this `| undefined ` is redundant
```suggestion
  tooltipContent?: string | null
```